apps/system/js/updatable.js

```diff
@@ -70,16 +91,103 @@ AppUpdatable.prototype.clean = function() {
 AppUpdatable.prototype.availableCallBack = function() {
   this.size = this.app.downloadSize;
 
-  if (this.app.installState === 'installed') {
-    UpdateManager.addToUpdatesQueue(this);
-
-    // we add these callbacks only now to prevent interfering
-    // with other modules (especially the AppInstallManager)
-    this.app.ondownloaderror = this.errorCallBack.bind(this);
-    this.app.ondownloadsuccess = this.successCallBack.bind(this);
-    this.app.ondownloadapplied = this.appliedCallBack.bind(this);
-    this.app.onprogress = this.progressCallBack.bind(this);
+  if (this.app.installState !== 'installed') {
+    return;
   }
+
+  let settings = [
+    setting(AppUpdatable.AUTO_UPDATE),
+    setting(AppUpdatable.NOTIFY_UPDATE)
+  ];
```

Comment:
We shouldn't read these settings each time. Better would be to read the setting the first time by adding a SettingsObserver, and then keep the value cached in a static variable.